isis/src/base/apps/enlarge/enlarge_app.cpp

```diff
@@ -0,0 +1,99 @@
+#include "enlarge_app.h"
+
+#include "CubeAttribute.h"
+#include "Enlarge.h"
+#include "IException.h"
+#include "ProcessRubberSheet.h"
+
+using namespace std;
+using namespace Isis;
+namespace Isis{
+
+  void enlarge(UserInterface &ui, Pvl *log) {
+    Cube *cube = new Cube(ui.GetFileName("FROM"), "r");
+    CubeAttributeInput inAtt = ui.GetInputAttribute("FROM");
+    if (inAtt.bands().size() != 0) {
+      cube->setVirtualBands(inAtt.bands());
+    }
```

Comment:
You have to set the virtual bands before you open the cube. You need to make a new Cube object without passing a filename, set the virtual bands, and then open the cube.